public/app/plugins/datasource/grafana-azure-monitor-datasource/datasource.ts

```diff
@@ -2,72 +2,105 @@ import _ from 'lodash';
 import AzureMonitorDatasource from './azure_monitor/azure_monitor_datasource';
 import AppInsightsDatasource from './app_insights/app_insights_datasource';
 import AzureLogAnalyticsDatasource from './azure_log_analytics/azure_log_analytics_datasource';
-import { AzureMonitorQuery, AzureDataSourceJsonData } from './types';
+import { AzureMonitorQuery, AzureDataSourceJsonData, AzureQueryType, InsightsAnalyticsQuery } from './types';
 import {
   DataSourceApi,
   DataQueryRequest,
   DataSourceInstanceSettings,
   DataQueryResponse,
   DataQueryResponseData,
+  LoadingState,
 } from '@grafana/data';
-import { Observable } from 'rxjs';
+import { Observable, of } from 'rxjs';
+import { DataSourceWithBackend } from '@grafana/runtime';
+import InsightsAnalyticsDatasource from './insights_analytics/insights_analytics_datasource';
 
 export default class Datasource extends DataSourceApi<AzureMonitorQuery, AzureDataSourceJsonData> {
   azureMonitorDatasource: AzureMonitorDatasource;
   appInsightsDatasource: AppInsightsDatasource;
   azureLogAnalyticsDatasource: AzureLogAnalyticsDatasource;
+  insightsAnalyticsDatasource: InsightsAnalyticsDatasource;
+
+  pseudoDatasource: Record<AzureQueryType, DataSourceWithBackend>;
+  optionsKey: Record<AzureQueryType, any>;
```

Comment:
nit: should be string I think?
```suggestion
  optionsKey: Record<AzureQueryType, string>;
```